ui/app/adapters/kv/metadata.js

```diff
@@ -68,11 +68,17 @@ export default class KvMetadataAdapter extends ApplicationAdapter {
     });
   }
 
-  // This method is only called when deleting from the LIST view. Otherwise, delete on kv/data
+  // This method is called when deleting from the list or metadata details view.
+  // Otherwise, delete happens in kv/data adapter
   deleteRecord(store, type, snapshot) {
     const { backend, path, fullSecretPath } = snapshot.record;
     // fullSecretPath is used when deleting from the LIST view and is defined via the serializer
     // path is used when deleting from the metadata details view.
     return this.ajax(this._url(kvMetadataPath(backend, fullSecretPath || path)), 'DELETE');
   }
+
+  // custom method when users do not permissions to fetch record
+  deleteAllVersions(backend, path) {
```

Comment:
I do not understand how the delete tests were passing before 🤯 🤯 🤯  For the secret-creator persona there was no model (because the user can delete but not read) Ember magic??

Reply:
NIT: there is another API endpoint that is used to technically delete all versions — granted you have to pass an  an array of all versions. But the naming here threw me. For me `deleteMetadata` makes the most sense as it's the delete capability on the metadata endpoint. I know in the docs they say "Delete metadata and all versions" but we do have designs out there as a way to delete all versions using the ["Delete secret versions"](https://developer.hashicorp.com/vault/api-docs/secret/kv/kv-v2#delete-secret-versions) endpoint.

Reply:
I can rename! Doesn't matter to me 😄